enhancements/machine-api/control-plane-set.md

```diff
@@ -0,0 +1,585 @@
+---
+title: control-plane-set-controller
+authors:
+  - "@JoelSpeed"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+api-approvers:
+  - TBD
+creation-date: 2022-01-11
+last-updated: 2022-01-17
+tracking-lin:
+  - TBD
+replaces:
+  - "[/enhancements/machine-api/control-plane-machinesets.md](https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/control-plane-machinesets.md)"
+  - "https://github.com/openshift/enhancements/pull/278"
+  - "https://github.com/openshift/enhancements/pull/292"
+---
+
+# Control Plane Set Controller
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+To enable automated management of vertical scaling and replacement of Control Plane Machines, this proposal introduces
+a new resource and controller that will manage Control Plane Machine instances.
+
+## Motivation
+
+As OpenShift adoption increases and our managed services offerings become more popular, the manual effort required to
+scale Control Plane Machines as clusters grow or shrink is becoming a significant drain on the OpenShift Dedicated team.
+
+The procedure to resize a Control Plane today is lengthy and very involved. It takes a significant amount of time for
+an OpenShift expert to perform. We also document this process for our end users, however due to the complexity of the
+procedure, there is discomfort in recommending this procedure to end users who may not be as familiar with the product.
+
+To ensure the long term usability of OpenShift, we must provide an automated way for users to scale their Control Plane
+as and when their cluster needs additional capacity.
+
+### Goals
+
+* Provide a "best practices" approach to declarative management of Control Plane Machines
+* Allow users to "1-click" scale their Control Plane to large or smaller instances
```

Comment:
As I understood it from initial requirements gather calls, yes. We need to scale up the control plane to bigger instances when the capacity is being stretched and be able to scale down to smaller instances when it is being under-utilized.

Having any kind of code check that the user is scaling down the instance type sounds like it will be very complex, we will have to start interpreting VM types which is not something we've done before

@jeremyeder Could you weigh in here